src/main/java/frc/utils/pose/PoseUtils.java

```diff
@@ -45,6 +45,43 @@ public static Pose2d poseArrayToPose2D(double[] poseArray, AngleUnit angleUnit)
 		);
 	}
 
+	public static double[] pose2DToPoseArray(Pose2d pose2d, AngleUnit angleUnit) {
+		return new double[] {pose2d.getX(), pose2d.getY(), switch (angleUnit) {
+			case RADIANS -> pose2d.getRotation().getRadians();
+			case DEGREES -> pose2d.getRotation().getDegrees();
+			case ROTATIONS -> pose2d.getRotation().getRotations();
+		}};
+	}
+
+	public static double[] pose3DToPoseArray(Pose3d pose3d, AngleUnit angleUnit) {
+		return switch (angleUnit) {
```

Comment:
```suggestion
	public static double[] pose3DToPoseArray(Pose3d pose3d, AngleUnit angleUnit) {
		Rotation3d angle = angleUnit.toRotation3d(Pose3d.getRotation())
		return switch (angleUnit) {
```

you'll need to create toRoation3d but it'll be shorter than this switch-case. 

ik you asked me about that, at the time I wasn't aware of that.